modules/price/src/Element/Price.php

```diff
@@ -61,7 +61,7 @@ public function getInfo() {
    * {@inheritdoc}
    */
   public static function valueCallback(&$element, $input, FormStateInterface $form_state) {
-    if ($input !== FALSE && $input !== NULL) {
+    if ($input !== FALSE && $input !== NULL && !empty($input['amount']) && !empty($input['currency_code'])) {
       // Convert empty string value to numeric value.
       if ($input['amount'] === '') {
```

Comment:
This conversion no longer makes sense if we pass on empty amounts? And won't this ignore a 0 EUR price, which is valid?